dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/registry/ServerNodeManager.java

```diff
@@ -235,12 +235,13 @@ public void notify(Event event) {
     }
 
     private void updateMasterNodes() {
-        currentSlot = 0;
-        totalSlot = 0;
-        this.masterNodes.clear();
+
         String nodeLock = Constants.REGISTRY_DOLPHINSCHEDULER_LOCK_MASTERS;
         try {
             registryClient.getLock(nodeLock);
+            currentSlot = 0;
```

Comment:
It's better to store the pre value of `currentSlot`, totalSlot, masterNodes, and rollback to the pre value when catch an exception.

Reply:
for what reson?

Reply:
> It's better to store the pre value of `currentSlot`, totalSlot, masterNodes, and rollback to the pre value when catch an exception.

for what reason?

Reply:
When this step fails it should rollback to the pre-state.